### PR TITLE
[NUI] Dialog code refactoring

### PR DESCRIPTION
**Pull request by jaehyun0cho**
### Description of Change ###

[NUI] Do not show navigation transition effect for DialogPage

The current navigation transition effect is not displayed when
DialogPage is pushed or popped.

[NUI] Set EnableDismissOnScrim true by default

To pop DialogPage when DialogPage's Scrim is touched by default,
EnableDismissOnScrim is set to be true by default.

[NUI] Add ScrimColor to DialogPage

To set color of Scrim, ScrimColor is added to DialogPage.

### API Changes ###

 - ACR:

### CODE REVIEW

**Comment by jaehyun0cho** [comment]
This PR should be merged by "Rebase and merge".

**Comment by jaehyun0cho** [comment]
This PR is based on https://github.com/Samsung/TizenFX/pull/2891

File: src/Tizen.NUI.Components/Controls/Navigation/DialogPage.cs [hunk 1]
```diff
@@ -198,7 +198,26 @@ public bool EnableScrim
         /// Indicates to dismiss dialog by touching on scrim.
         /// </summary>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public bool EnableDismissOnScrim { get; set; }
+        public bool EnableDismissOnScrim { get; set; } = true;
```
Comment (dongsug-song) [inline src/Tizen.NUI.Components/Controls/Navigation/DialogPage.cs:201]: How about `DismissOnScrim`?
`EnableDismissOnScrim` seems to be a name of method.

**Review by dongsug-song — APPROVED** [review]
looks good to me

**Pull request merged**